### PR TITLE
Refactor resource util to match specs.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Blocked by https://github.com/census-instrumentation/opencensus-specs/pull/235

### CODE REVIEW

File: contrib/resource_util/src/main/java/io/opencensus/contrib/resource/util/CloudResource.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          PROVIDER_AWS,
+          AwsIdentityDocUtils.getAccountId(),
+          AwsIdentityDocUtils.getRegion(),
+          AwsIdentityDocUtils.getAvailabilityZone());
+    }
+    if (GcpMetadataConfig.isRunningOnGcp()) {
+      return create(
+          PROVIDER_GCP, GcpMetadataConfig.getProjectId(), "", GcpMetadataConfig.getZone());
+    }
+    return EMPTY_RESOURCE;
```
Comment (songy23) [inline contrib/resource_util/src/main/java/io/opencensus/contrib/resource/util/CloudResource.java]: nit: consider leaving a TODO for detecting Azure.

File: contrib/resource_util/src/main/java/io/opencensus/contrib/resource/util/ContainerResource.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  public static Resource create(String name, String imageName, String imageTag) {
+    Map<String, String> labels = new LinkedHashMap<String, String>();
+    labels.put(NAME_KEY, checkNotNull(name, "name"));
+    labels.put(IMAGE_NAME_KEY, checkNotNull(imageName, "imageName"));
+    labels.put(IMAGE_TAG_KEY, checkNotNull(imageTag, "imageTag"));
+    return Resource.create(TYPE, labels);
+  }
+
+  static Resource detect() {
+    return create(firstNonNull(System.getenv("CONTAINER_NAME"), ""), "", "");
```
Comment (songy23) [inline contrib/resource_util/src/main/java/io/opencensus/contrib/resource/util/ContainerResource.java]: Is there a way to detect image name and tag? (maybe leave a TODO here?)

Comment (bogdandrutu) [reply contrib/resource_util/src/main/java/io/opencensus/contrib/resource/util/ContainerResource.java]: done

File: contrib/resource_util/src/main/java/io/opencensus/contrib/resource/util/CloudResource.java (hunk 1 above)
Comment (bogdandrutu) [reply contrib/resource_util/src/main/java/io/opencensus/contrib/resource/util/CloudResource.java]: done.

File: exporters/trace/stackdriver/src/test/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandlerProtoTest.java [hunk 3]
```diff
@@ -130,38 +129,24 @@
...
-  private static final ImmutableMap<String, AttributeValue> EC2_RESOURCE_LABELS =
-      ImmutableMap.of(
-          createResourceLabelKey(AwsEc2InstanceResource.ACCOUNT_ID_KEY),
-          toStringAttributeValueProto("my-project"),
-          createResourceLabelKey(AwsEc2InstanceResource.REGION_KEY),
-          toStringAttributeValueProto("us-east-1"),
-          createResourceLabelKey(AwsEc2InstanceResource.INSTANCE_ID_KEY),
-          toStringAttributeValueProto("my-instance"));
-  private static final ImmutableMap<String, AttributeValue> GCE_RESOURCE_LABELS =
+  private static final Resource CUSTOM_RESOURCE =
```
Comment (songy23) [inline exporters/trace/stackdriver/src/test/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandlerProtoTest.java]: Just to double check: for Stackdriver Trace exporter we don't need to do resource label mappings? (use the modified labels as-is?)

**Comment by songy23** [comment]
Is this ready to merge?

**Comment by bogdandrutu** [comment]
Thanks for merging this.

**Pull request merged**